services/src/main/scala/cromwell/services/metadata/impl/archiver/ArchiveMetadataSchedulerActor.scala

```diff
@@ -130,48 +132,57 @@ class ArchiveMetadataSchedulerActor(archiveMetadataConfig: ArchiveMetadataConfig
       workflowSelectedTime = OffsetDateTime.now()
-      _ = sendTiming(archiverTimingMetricsBasePath :+ "lookup_next_workflow", calculateTimeDifference(startTime, workflowSelectedTime), ServicesPrefix)
-
-      result <- maybeWorkflowSummaryEntry match {
-        case Some(summaryEntry) =>
-          summaryEntry.endTimestamp.foreach { workflowEndTime =>
-            val millisSinceWorkflowEnd = JDuration.between(workflowEndTime.toSystemOffsetDateTime, OffsetDateTime.now()).toMillis
-            sendGauge(timeBehindExpectedDelayMetricPath, millisSinceWorkflowEnd - archiveMetadataConfig.archiveDelay.toMillis, ServicesPrefix)
-          }
-          for {
-            path <- Future.fromTry(getGcsPathForMetadata(summaryEntry))
-            workflowId = summaryEntry.workflowExecutionUuid
-            dbStream <- fetchStreamFromDatabase(WorkflowId(UUID.fromString(workflowId)))
-            _ = log.info(s"Archiving metadata for $workflowId to ${path.pathAsString}")
-            readyToStreamTime = OffsetDateTime.now()
-            _ = sendTiming(archiverTimingMetricsBasePath :+ "prepare_to_stream", calculateTimeDifference(workflowSelectedTime, readyToStreamTime), ServicesPrefix)
-            _ <- streamMetadataToGcs(path, dbStream)
-            streamCompleteTime = OffsetDateTime.now()
-            _ = sendTiming(archiverTimingMetricsBasePath :+ "stream_to_gcs", calculateTimeDifference(readyToStreamTime, streamCompleteTime), ServicesPrefix)
-            _ <- updateMetadataArchiveStatus(WorkflowId(UUID.fromString(workflowId)), Archived)
-            statusUpdatedTime = OffsetDateTime.now()
-            _ = sendTiming(archiverTimingMetricsBasePath :+ "archive_status_update", calculateTimeDifference(streamCompleteTime, statusUpdatedTime), ServicesPrefix)
-            _ = log.info(s"Archiving succeeded for $workflowId")
-          } yield true
-        case None =>
-          sendGauge(timeBehindExpectedDelayMetricPath, 0L, ServicesPrefix)
-          Future.successful(false)
-      }
+      _ = sendTiming(archiverTimingMetricsBasePath :+ "lookup_next_workflows", calculateTimeDifference(startTime, workflowSelectedTime), ServicesPrefix)
+      _ = log.info(s"About to archive batch of ${workflowSummaryEntries.size} workflows.")
+      result <- archiveSummaryEntries(workflowSummaryEntries, workflowSelectedTime)
     } yield result
   }
 
-  def lookupNextWorkflowToArchive(): Future[Option[WorkflowMetadataSummaryEntry]] = {
+  private def archiveSummaryEntries(entries: Seq[WorkflowMetadataSummaryEntry], workflowSelectedTime: OffsetDateTime): Future[Long] = {
+    if (entries.isEmpty) {
+      sendGauge(timeBehindExpectedDelayMetricPath, 0L, ServicesPrefix)
+      Future.successful(0)
+    } else {
+      val resultSeq: Seq[Future[Long]] = entries.map(archiveSummaryEntry(_, workflowSelectedTime))
+      val result: Future[Seq[Long]] = Future.sequence(resultSeq) // The special sauce that makes 'em execute in parallel
```

Comment:
Trust but verify... but I have a feeling that the `map + sequence` here is equivalent to `traverse`

Reply:
That is accurate, the following is the one-line equivalent of my code. But I just feel like it's easier to understand when expanded.
```
Future.traverse(entries)(archiveSummaryEntry(_, workflowSelectedTime)).map(_.sum)
```